xdsl/dialects/builtin.py

```diff
@@ -313,21 +352,39 @@ class FloatAttr(Generic[_FloatAttrTyp], ParametrizedAttribute):
     value: ParameterDef[FloatData]
     type: ParameterDef[_FloatAttrTyp]
 
-    @staticmethod
-    def from_value(
-        value: float, type: _FloatAttrTypContr = Float32Type()
-    ) -> FloatAttr[_FloatAttrTypContr]:
-        return FloatAttr([FloatData.from_float(value), type])
-
-    @staticmethod
+    @overload
+    def __init__(self, data: float | FloatData, type: _FloatAttrTyp) -> None:
+        ...
+
+    @overload
+    def __init__(self, data: float | FloatData, type: int) -> None:
+        ...
+
+    def __init__(self, data: float | FloatData,
+                 type: int | _FloatAttrTyp | AnyFloat) -> None:
```

Comment:
the `-> None` here feels so weird, but I guess that's how it's done ([pep](https://peps.python.org/pep-0484/)). Pyright doesn't allow these to be on-annotated, right? If it does, I would prefer it over the None annotation.

Reply:
Unannotated means "infer", which is less informative than an explicit `None`

Reply:
Hmm okay. I guess that's just the way it's done then...